_P164_MSswitch.ino

```diff
@@ -0,0 +1,169 @@
+#ifdef USES_P164
+//#######################################################################################################
+//#################################### Plugin 164: MSswitch #############################################
+//#######################################################################################################
+
+// Februray 3, 2019 initial release
+// 433MHZ Transmitters: http://www.rflink.nl/blog2/wiring
+// Known supported auto-learning switches are Nexa, Prove, Telldus & Jula Anslutt
+
+// List of commands:
+// (1) MSswitch,<G,1-16>,<ON/OFF>
+
+// Usage:
+// (1): Turn on switches in group (eg. MSswitch,G,on)
+
+#include <Socketswitch.h>
+
+// Download library from https://github.com/mortensalomon/Socket_Switch
+
+Socketswitch *Plugin_164_msswitch;
+
+#define PLUGIN_164
+#define PLUGIN_ID_164         99
+#define PLUGIN_NAME_164       "MS auto-learning Switch"
+#define PLUGIN_VALUENAME1_164 ""
+
+unsigned long SocketAddress;
+String log164;
+
+boolean Plugin_164(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number = PLUGIN_ID_164;
+      Device[deviceCount].Type = DEVICE_TYPE_SINGLE;
+      Device[deviceCount].Ports = 0;
+      Device[deviceCount].Custom = true;
+      Device[deviceCount].TimerOption = false;
+      Device[deviceCount].SendDataOption = false;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_164);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_164));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      addRowLabel(F("433MHZ Data Pin"));
+      addPinSelect(false, F("taskdevicepin1"), CONFIG_PIN1);
+      addFormNumericBox(F("Switch Address"), F("p164_address"), PCONFIG_LONG(0),0,67108863);
+
+      if (CONFIG_PIN1>-1) {
+        addFormSubHeader(F("Try it"));
+        String options[17] = { F("CH 1"),F("CH 2"),F("CH 3"),F("CH 4"),F("CH 5"),F("CH 6"),F("CH 7"),F("CH 8"),F("CH 9"),F("CH 10"),F("CH 11"),F("CH 12"),F("CH 13"),F("CH 14"),F("CH 15"),F("CH 16"),F("GROUP") };
+        int optionValues[17] = { 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16 };
```

Comment:
Why not use a for loop for this?
These strings also add to the size of the binary.